## Go
만들고 있는 고랭 프로젝트에서 validator 를 만드는데 패닉이 발생해서 체크해 보니 에러 코드 처리에서 문제가 있었다. 코드 넘버를 그대로 사용하는 이상한 짓을 하고 있었다.

개편한 코드는 아래와 같다.
```go

type CustomErr struct {
	StatusCode int
	Code       int
	Msg        string
}

```
커스텀 에러 타입에 StatusCode를 추가하고

```go
var (

	// == set == //
	ERR_BINDING   = CustomErr{StatusCode: http.StatusBadRequest, Code: 000, Msg: "request binding err"}
	ERR_VALIDATOR = CustomErr{StatusCode: http.StatusBadRequest, Code: 001, Msg: "request validator err"}
)
```
상수로 에러 코드를 관리하면서 http 응답값을 적어주었다.

```go
var req dto.CreateUserReq
	if err := validator.BinderAndValidator(g, &req); err != nil {
		g.JSON(err.StatusCode, err)
		return
	}
```
controller에서는 객체 바인딩과 동시에 유효성 검증을 진행하고 status code를 실어서 리턴해주었다.

기존에는 커스텀 부분에서 커스텀 코드 자체를 리턴해주고 있었기 때문에 001에서 패닉이 발생했던 것 같다.